source/server/converters.h

```diff
@@ -317,6 +318,21 @@ typename TypeToStorageType<TDriverType, TGrpcType>::StorageType allocate_output_
 }
 
 }  // namespace converters
+
+const int kMaxGrpcErrorDescriptionSize = 4096;
```

Comment:
I wasn't aware of this before, but https://cloud.google.com/apis/design/errors#error_model has a specific guideline about error size:

> Note: When using gRPC, errors are included in the headers, and total headers in responses are limited to 8 KB (8,192 bytes). Ensure that errors do not exceed 1-2 KB in size.

Reply:
What are the situations in which it will vary? I tested local servers for both C# and Python and 4K came through just fine. I know NI has error messages longer than 1KB but do we have any longer than 2KB?

Reply:
I suspect that this guideline errs on the side of caution, in order to leave room for other response trailers. Also, I think the "richer error model" sends the status code and message twice: once as separate `grpc-status` and `grpc-message` response trailers, and again as fields in the `google.rpc.Status` message that is sent in the `grpc-status-details-bin` response trailer.

> I know NI has error messages longer than 1KB but do we have any longer than 2KB?

Static error messages: I don't know.

Dynamic error messages:
- RF/MI: as we discussed previously, these drivers may or may not truncate the dynamic part of the error messages to 1024 bytes, depending on which implementation of `GetError` is called. 
- DAQmx: I don't think the DAQmx C API has a hard limit, but nidaqmx-python truncates error messages to 2 KB: https://github.com/ni/nidaqmx-python/blob/master/nidaqmx/errors.py#L182